internal/internal_utils.go

```diff
@@ -226,8 +226,12 @@ func getErrorDetails(err error, dataConverter DataConverter) (string, []byte) {
 // constructError construct error from reason and details sending down from server.
 func constructError(reason string, details []byte, dataConverter DataConverter) error {
 	if strings.HasPrefix(reason, errReasonTimeout) {
-		timeoutType := getTimeoutTypeFromErrReason(reason)
 		details := newEncodedValues(details, dataConverter)
+		timeoutType, err := getTimeoutTypeFromErrReason(reason)
+		if err != nil {
+			details.Get(&timeoutType)
```

Comment:
Doesn't this call return error ?

Reply:
Good point. I've update the code to check the error. If the error is not nil, I think the only thing I can do is panic. 

I also checked old versions of the code, it seems like this error is never handled. I am not sure if that's deliberate or not. Maybe we want to avoid panic and use the default value as timeout type?